legacy-migration/providers/migrate.py

```diff
@@ -0,0 +1,312 @@
+"""Migrates providers from legacy services REST API into new providers interface"""
+import json
+import os
+import xml.etree.ElementTree as ET
+import re
+import time
+import sys
+import requests
+
+URL_ROOT = "https://cmr.sit.earthdata.nasa.gov"
+
+access_token = os.environ.get('SIT_SYSTEM_TOKEN')
+CMR_INGEST_ENDPOINT = "https://cmr.sit.earthdata.nasa.gov/ingest"
+
+# To test local ingesting into CMR
+local_cmr_access_token = os.environ.get('localHeader')
+# These are for the Enum values that we must set in the metadata field
+# TODO I want more clarity on how to handle these organization and contact roles since there is not an enum accessible substitute in the old format
+DEFAULT_ORGANIZATION_ROLE = "SERVICE PROVIDER"
+DEFAULT_CONTACT_ROLE = "TECHNICAL CONTACT"
+
+# Metadata specification for ingesting providers
+# It should not be expected this version or schema directories will change
+# within the lifecycle of this migration
+metadata_specification = {
+            "Name": "provider",
+            "Version": "1.0.0",
+            "URL": "https://cdn.earthdata.nasa.gov/schemas/provider/v1.0.0"}
+
+def ingest_provider(provider_metadata, provider_id):
+    """Attempts to ingest metadata into CMR (locally)"""
+    response = requests.post("http://localhost:3002/providers/",
```

Comment:
Set both CMR client and browser agent in all scripts you write, they can be the same value.